sdk/python/kubeflow/training/api/training_client.py

```diff
@@ -36,35 +36,43 @@
 class TrainingClient(object):
     def __init__(
         self,
-        config_file=None,
-        context=None,
-        client_configuration=None,
-        persist_config=True,
+        config_file: str = None,
+        context: str = None,
+        client_configuration: client.Configuration = None,
+        no_config: bool = False,
```

Comment:
This is not necessary? If `client_configuration` is not None, then it should take precedence over `config_file`

Reply:
If we remove `no_config`, I guess we can not inform about missing `client_configuration `, right?
WDYT?

Reply:
@tenzen-y is right . I did this since user can set `config_file` and `client_configuration` to add additional information to the config at the same time.

Reply:
We had discussion with @johnugeorge around this.

I guess kube-config supports all configuration that user can set via `client.Configuration()` object (proxy, token. etc.).
@terrytangyuan @tenzen-y Did you ever see the use-case when user wants to set custom `client_configuration` with existing kubeconfig ?

If not, we can do this:
- If user set `config_file` use `load_kube_config()`.
- If user doesn't set `config_file` and doesn't set `client_configuration` use `load_kube_config()` or `load_incluster_config()`.
- If user set `client_configuration` skip `load_kube_config()` and `load_incluster_config()`.


`no_config` flag might look a bit hacky.

Reply:
No, I have not seen that use case.